### PR TITLE
[DA-127]: Integrate with Repour to clone the GIT repository

### PR DESCRIPTION
**Pull request by jbartece**

### CODE REVIEW

File: bc-backend/src/main/java/org/jboss/da/bc/backend/impl/RepositoryClonerImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Service to clone the repositories to the internal systems
+ * 
+ * @author Jakub Bartecek <jbartece@redhat.com>
+ *
+ */
+@ApplicationScoped
+public class RepositoryClonerImpl implements RepositoryCloner {
+
+    private static final String REPOUR_BASE_URL = "http://ncl-test-vm-02.host.prod.eng.bos.redhat.com:7331";
```
Comment (thescouser89) [inline bc-backend/src/main/java/org/jboss/da/bc/backend/impl/RepositoryClonerImpl.java]: Maybe move it to the config file?

File: bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepositoryCloner.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     * Clones the repository to the internal systems. Source repository has to be available without authentication
+     * 
+     * @param url URL to the source repository
+     * @param revision Revision, which will be cloned
+     * @param scmType Type of the repository. Currently is supported only GIT
+     * @param repositoryName Name of the new repository
+     * @return URL to the new repository
+     * @throws Exception Thrown if the cloning process failed
+     */
+    String cloneRepository(String url, String revision, SCMType scmType, String repositoryName)
```
Comment (thescouser89) [inline bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepositoryCloner.java]: From the recent conversation I had with Alex S and how repour does cloning, is revision a tag, commit id, or can be both?

**Comment by janinko** [comment]
:+1:

File: bc-backend/src/main/java/org/jboss/da/bc/backend/impl/RepositoryClonerImpl.java (hunk 1 above)
Comment (jbartece) [inline bc-backend/src/main/java/org/jboss/da/bc/backend/impl/RepositoryClonerImpl.java]: I'll do that

File: bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepositoryCloner.java (hunk 2 above)
Comment (jbartece) [inline bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepositoryCloner.java]: I tested it with branches, but I have to check it with Alex.

**Comment by jbartece** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

**Pull request merged**